### PR TITLE
Dividing Task to Student [CP][C++]

### LINKED ISSUE #5456: Dividing Tasks to student [CP][GSSOC'21]
## 💥 Proposal

Dividing Tasks to student is CP problem that state that  There are N students in the class, i'th student has Ci task in his collection. Your task is to determine if it is possible to redistribute C1 + C2 + ... + Cn task among the student of class thus that i'th student would get i task.

[GSSOC'21][CP][C-Plus-Plus]

### Have you read the [Contributing Guidelines on Pull Requests][ ](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?
Yes
(Write your answer here.)

### PR DESCRIPTION
**Pull request by vedant-jain03**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Description

(Dividing Tasks to student is CP problem that state that There are N students in the class, i'th student has Ci task in his collection. Your task is to determine if it is possible to redistribute C1 + C2 + ... + Cn task among the student of class thus that i'th student would get i task.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

fixes #5456

### CODE REVIEW

**Comment by vedant-jain03** [comment]
@ankitaggarwal23 @HarshCasper 
Please review my code and issue

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -124,6 +124,7 @@
 - [Inversion Count Problem using Merge Sort](cp/InversionCountProblem.cpp)
 - [Minimum circles to be removed](cp/min_circles_to_be_removed.cpp)
 - [Busy Man](cp/busy_man.cpp)
+- [Divind Task to Students](cp/Dividing_task.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md]: Typo here

File: C-Plus-Plus/cp/Dividing_task.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+// given number of students (n)
+// given the task assigned to them
+// output if the problem state is possible if true print Yes else No
+
+// Algorithm
+// declare and initialise sum_of_task that means sum of task given in array to 0
+// declare and initialise sum_of_i_student means sum of ith student to 0
+// if sum_of_task is equal to sum_of_i_student is equal print YES else NO
+#include <bits/stdc++.h>
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Dividing_task.cpp]: Use multi-line comments and add a more intuitive documentation here

File: C-Plus-Plus/cp/Dividing_task.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+// 5
+// 7 4 1 1 2
+// output
+// YES
+
+// Test case 2
+// 5
+// 1 1 1 1 1
+// output
+// NO
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Dividing_task.cpp]: Use Multi-Line Comments

File: C-Plus-Plus/cp/Dividing_task.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// given the task assigned to them
+// output if the problem state is possible if true print Yes else No
+
+// Algorithm
+// declare and initialise sum_of_task that means sum of task given in array to 0
+// declare and initialise sum_of_i_student means sum of ith student to 0
+// if sum_of_task is equal to sum_of_i_student is equal print YES else NO
+#include <bits/stdc++.h>
+using namespace std;
+int main()
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Dividing_task.cpp:19]: Don't put the code logic in `main()` function. Keep it in another function and use `main()` for the driver code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by vedant-jain03 — COMMENTED** [review]
Resolvd Issue and commit

**Comment by vedant-jain03** [comment]
@HarshCasper Review my Changes
@ritvij14 @anushkrishnav

File: C-Plus-Plus/cp/Dividing_task.cpp [hunk 5]
```diff
@@ -0,0 +1,46 @@
+/*
+Problem Desciption:
+The problem states that, there are N students in the class, i'th student has assign Ci number of task. Your task is to determine if it is possible to redistribute C1 + C2 + ... + Cn task among the student of class thus that i'th student would get i task.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Dividing_task.cpp]: Modularize this statement to make it crisp and easy to understand.

File: C-Plus-Plus/cp/Dividing_task.cpp [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        cin>>a;
+        arr.push_back(a);
+        sum_of_task=sum_of_task+a;
+    }
+    if(sum_of_i_student==sum_of_task)cout<<"YES\n";
+    else cout<<"NO\n";
+}
+int main()
+{
+    int t;
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Dividing_task.cpp]: No need of test cases here. Only one sample I/O is enough

File: C-Plus-Plus/cp/Dividing_task.cpp [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    for(int i=1;i<=n;i++)
+    {
+        sum_of_i_student=sum_of_i_student+i;
+        int a;
+        cin>>a;
+        arr.push_back(a);
+        sum_of_task=sum_of_task+a;
+    }
+    if(sum_of_i_student==sum_of_task)cout<<"YES\n";
+    else cout<<"NO\n";
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Dividing_task.cpp]: Format the Code

File: C-Plus-Plus/cp/Dividing_task.cpp [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+Test case :
+2
+5
+7 4 1 1 2
+5
+1 1 1 1 1
+output
+YES
+NO
+*/
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Dividing_task.cpp]: Add a new line at the EOF

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by vedant-jain03 — COMMENTED** [review]
Code updated and formatted

**Pull request closed without merging**